File: DevEnvironment/Base/src/main/java/base/core/OperatingSystemControlSettings.java
package base.core;

import base.interfaces.IOperatingSystemControlSettings;
import base.interfaces.IOperatingSystemPathSettings;

/**
 * The OperatingSystemControlSettings class implements a control settings of an operating system.
 */
public final class OperatingSystemControlSettings implements IOperatingSystemControlSettings {
    private final IOperatingSystemPathSettings pathSettings;

    /**
     * The OperatingSystemControlSettings constructor.
     */
    public OperatingSystemControlSettings(IOperatingSystemPathSettings pathSettings) {
        Conditions.validateNotNull(
            pathSettings,
            "The path settings.");

        this.pathSettings = pathSettings;
    }

    /**
     * Gets paths settings.
     */
    @Override
    public IOperatingSystemPathSettings pathSettings() {
        return this.pathSettings;
    }
}

File: SystemCore/Base/core/OperatingSystemControlSettings.cpp
#include "OperatingSystemControlSettings.h"
#include "SmartPointers.h"

using namespace base::environment;

/**
 * The OperatingSystemControlSettings constructor.
 */
OperatingSystemControlSettings::OperatingSystemControlSettings(const IOperatingSystemPathSettingsSharedPtr pathSettings) :
    m_pathSettings(SmartPointers::ValidateAndReturn(pathSettings))
{
}

/**
 * The OperatingSystemControlSettings destructor.
 */
OperatingSystemControlSettings::~OperatingSystemControlSettings()
{
}

/**
 * Gets paths settings.
 */
IOperatingSystemPathSettingsSharedPtr OperatingSystemControlSettings::GetPathSettings() const
{
    return m_pathSettings;
}
